### PR TITLE
allow querying by fieldmanager

### PR DESCRIPTION
**Pull request by deads2k**
Add inspection by field manager.

/assign @bertinatto @p0lyn0mial

### CODE REVIEW

File: pkg/cmd/audit/audit_filter.go [hunk 1]
```diff
@@ -177,6 +188,29 @@ func (f *FilterByResources) Matches(event *auditv1.Event) bool {
...
+	}
+	return username
+}
+
+func QueryParams(uri string) url.Values {
+	// some request URL has query parameters like: /apis/image.openshift.io/v1/images?limit=500&resourceVersion=0
+	// we are not interested in the query parameters.
+	parts := strings.Split(uri, "?")
+	if len(parts) < 2 {
+		return nil
```
Comment (p0lyn0mial) [inline pkg/cmd/audit/audit_filter.go:204]: won't this blow up?

File: pkg/cmd/audit/io.go [hunk 2]
```diff
@@ -122,7 +124,11 @@ func PrintTopByUserAuditEvents(writer io.Writer, numToDisplay int, events []*aud
 	countUsers := map[string][]*auditv1.Event{}
 
 	for _, event := range events {
-		countUsers[event.User.Username] = append(countUsers[event.User.Username], event)
+		username := event.User.Username
+		if fieldManager := QueryParams(event.RequestURI).Get("fieldManager"); len(fieldManager) > 0 {
+			username = fmt.Sprintf("%s[%s]", username, fieldManager)
```
Comment (p0lyn0mial) [inline pkg/cmd/audit/io.go:129]: won't this be also applied even if `field-manager` wasn't specified ?

File: pkg/cmd/audit/audit_filter.go (hunk 1 above)
Comment (deads2k) [reply pkg/cmd/audit/audit_filter.go:204]: no.  `url.Values` is a map and retrieval from a nil map is valid in golang.

File: pkg/cmd/audit/io.go (hunk 2 above)
Comment (deads2k) [reply pkg/cmd/audit/io.go:129]: I think if the `if fieldManager := QueryParams(event.RequestURI).Get("fieldManager"); len(fieldManager) > 0 {` works as I expect, this only shows up if the fieldManager is present.

**Pull request merged**